core/src/main/java/io/questdb/cairo/TableReader.java

```diff
@@ -1021,9 +1021,10 @@ private void reloadColumnAt(
                     long columnSize = columnRowCount * 8L + 8L;
                     TableUtils.iFile(path.trimTo(plen), name, columnTxn);
                     mem2 = openOrCreateMemory(path, columns, secondaryIndex, mem2, columnSize);
-                    columnSize = mem2.getLong(columnRowCount * 8L);
+                    long columnSize2 = mem2.getLong(columnRowCount * 8L);
+                    assert columnSize > 0 && columnSize <= (1 << 30);
                     TableUtils.dFile(path.trimTo(plen), name, columnTxn);
-                    openOrCreateMemory(path, columns, primaryIndex, mem1, columnSize);
+                    openOrCreateMemory(path, columns, primaryIndex, mem1, columnSize2);
```

Comment:
should be `column2Size`

Reply:
done in https://github.com/questdb/questdb/pull/3053/commits/59559afc40f4e8b8c93aefd1fba854143a5484ac